### PR TITLE
Add supporter required placeholder in beatmap listing overlay

### LINKED ISSUE #13530: Beatmap listing supporter-only filter show "...nope, nothing found." instead of telling you that you need an active osu!supporter tag
**Screenshots or videos showing encountered issue:** 
osu!web:
image

osu!lazer:
image

**osu!lazer version:** 
2021.612.0

### PR DESCRIPTION
**Pull request by JimmyC7834**
Fixes #13530

Added ```SupporterRequiredDrawable``` in ```osu.Game.Overlays.BeatmapListingOverlay``` for non-supporter users applied supporter-only filters, referencing [web version](https://osu.ppy.sh/beatmapsets?played=played&r=XH)'s layout

Changed ```osu.Game.Overlays.BeatmapListing.BeatmapListingFilterControl``` to pass ```searchControl``` in ```SearchFinished``` for text in ```SupporterRequiredDrawable```

Tests added in ```osu.Game.Tests.Visual.Online.TestSceneBeatmapListingOverlay```
Tested on:
- non-supporter user with/without supporter-only filters, with/without search results
- supporter user with/without supporter-only filters, with/without search results

### CODE REVIEW

File: osu.Game/Overlays/BeatmapListingOverlay.cs
Comment (bdach) [inline osu.Game/Overlays/BeatmapListingOverlay.cs:305]: I don't like this design as it doesn't scale (what if 3 more supporter-specific filters are added N months from now? are 3 more `bool`s going to be added?) This should really be a collection of some sort (not 100% sure of what exactly, the immediate candidate is a collection of the localisation keys themselves).

File: osu.Game/Overlays/BeatmapListingOverlay.cs
Comment (bdach) [inline osu.Game/Overlays/BeatmapListingOverlay.cs:315]: `LinkFlowContainer` also supports adding plain text. So both the supporter tag link and the preceding text should be in a `LinkFlowContainer`, and not by this current `SpriteText` + `LinkFlowContainer` setup.

File: osu.Game/Overlays/BeatmapListingOverlay.cs
Comment (bdach) [inline osu.Game/Overlays/BeatmapListingOverlay.cs:323]: * By convention, untranslatable string literals use the raw string notation (`@"https://osu.ppy.sh/store/products/supporter-tag"`).
* The URL of the website changes depending on debug/release configuration, so the `https://osu.ppy.sh` domain should not be hardcoded like here. Use `IAPIProvider.WebsiteRootUrl` to handle this properly.

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs
Comment (bdach) [inline osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs:29]: outdated xmldoc

File: osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs
Comment (bdach) [inline osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs:219]: This type of check will also not scale well if there are more supporter-only filters in the future. A simple change that would improve this would involve removing the `Ranks` and `Played` checks from this main `if`, and then doing

```csharp
    if (filters.Any())
    {
        SearchFinished?.Invoke(SearchResult.SupporterOnlyFilter(filters));
        return;
    }
} // end of supporter check block

SearchFinished?.Invoke(SearchResult.ResultsReturned(sets));
```
lower down.

File: osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs
Comment (bdach) [inline osu.Game/Overlays/BeatmapListing/BeatmapListingFilterControl.cs:266]: structure should have xmldoc, as I requested when suggesting it

File: osu.Game/Overlays/BeatmapListingOverlay.cs
Comment (bdach) [inline osu.Game/Overlays/BeatmapListingOverlay.cs:316]: This doesn't have to be done this way. `AddText()` has an overload that accepts a `SpriteText`.

I'd also probably just skip this and recreate the entire text on every change. It doesn't really matter all that much for now, but it might later. Putting an entire string with spaces in a single `SpriteText`, which is in a `TextFlowContainer` makes it so that that text won't... flow. As long as there are no overly long variations of this string, that would begin to overflow the view, I guess it's passable for now.

**Review by bdach — CHANGES_REQUESTED** [review]

**Comment by bdach** [comment]
I've applied further cleanups and think this mostly ready to go, but I am not going to merge this without a second approval because I'm not sure whether I'll get agreement on how the search result structure is set up. It makes sense to me but I do want to run that through another pair of eyes first.

**Review by bdach — APPROVED** [review]

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**
